StudCity.Frontend/src/pages/ViewProfile/index.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from "react";
+import HeaderViewProfile from "../../../components/HeaderViewProfile";
+import ElementViewProfile from "../../../components/ElementViewProfile";
+import Container from "../../../components/Container";
+import { useNavigate, useParams } from "react-router-dom";
+import { UserService } from "../../../services/userService";
+import { USER_PROFILE_ICONS } from "../../../utils/constants";
+import Spinner from "../../../components/Spinner";
+
+function ViewProfile() {
+  const service = new UserService();
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState({
+    userName: "USER_NAME",
+    email: "EMAIL",
+	dateOfBirthday: "DATE",
+    phoneNumber: "PHONE_NUMBER",
+    gender: "GENDER",
+    role: "ROLE"
+  })
+
+  const [otherInfo, setOtherInfo] = useState({
+    fullName: "Yaroslav Bihun",
+    avatar: "/images/student.jpg",
+    backgroundImage: "/images/FlagUkraine.jpeg",
+    description: "Cool guy"
+  });
+
+  useEffect(() => {
+    (async () => {
+      await fetchUser(id);
+    })()
+  }, [id])
+
+  const fetchUser = async (id) => {
+    setLoading(true);
+    const data = await service.getUserById(id);
+    if (!data) {
+      navigate("/not-found");
+      return;
+    }
+    setUser({
```

Comment:
I would suggest doing destructuring of data constant, and use a shorter syntax for setting a state:
const { userName, email, dateOfBirthday, phoneNumber,... } = data;
setUser({ userName, email, dateOfBirthday, .... });